pkg/controller/provider/controller.go

```diff
@@ -190,6 +190,31 @@ func (r Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (r
 		}
 	}
 
+	if provider.DeletionTimestamp != nil && k8sutil.ContainsFinalizer(provider, api.OvaProviderFinalizer) {
+		fmt.Println("we are here at finalizer")
+		labelSelector := labels.SelectorFromSet(labels.Set{
+			"subapp":   "ova-server",
+			"app":      "forklift",
+			"provider": provider.Name,
+		})
+		pvList := &v1.PersistentVolumeList{}
+		if err = r.Client.List(context.TODO(), pvList, &client.ListOptions{LabelSelector: labelSelector}); err != nil {
+			r.Log.Error(err, "Failed to list PVs for OVA provider", "provider", provider)
+		} else {
+			for _, pv := range pvList.Items {
+				if err = r.Client.Delete(context.TODO(), &pv); err != nil {
+					r.Log.Error(err, "Failed to delete PV", "PV", pv)
+					return
+				}
+			}
+			clonedProvider := provider.DeepCopy()
+			k8sutil.RemoveFinalizer(provider, api.OvaProviderFinalizer)
+			if err := r.Patch(context.TODO(), provider, client.MergeFrom(clonedProvider)); err != nil {
+				r.Log.Error(err, "Failed to remove finalizer", "provider", provider)
+			}
+		}
+	}
+
 	if provider.Type() == api.Ova {
```

Comment:
we shouldn't run this block when removing the provider
how about checking `provider.DeletionTimestamp == nil` and then we can keep the previous order?

Reply:
sure it will provide the same outcome, so i'm fine with both ways.